File: EatHealthy/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace EatHealthy.Models
{
    public class Comment
    {
        public int ID { get; set; }

        [Required]
        [ForeignKey("Client")]
        public int ClientID { get; set; }

        [Required]
        [ForeignKey("Post")]
        public int PostID { get; set; }

        [Required]
        public string Content { get; set; }

        [DataType(DataType.Date)]
        [DisplayName("Created at")]
        public DateTime CreationDate { get; set; }

        public virtual Post Post { get; set; }

        public virtual Client Client { get; set; }
    }
}

File: NewEatHealthy/server/models/Comment.js
var mongoose = require('mongoose')
  , Schema = mongoose.Schema
  , ObjectId = Schema.ObjectId;

  var CommentSchema = new Schema({
    content: {
        type :  String, required: true
    },
    creationDate : {
        type : Date, default: Date.now
    },
    post : {
        type: Schema.Types.ObjectId, ref: 'Post' 
    },
    client : {
        type: Schema.Types.ObjectId, ref: 'Client' 
    },
  });

exports.CommentSchema = CommentSchema;
module.exports = mongoose.model('Comment', CommentSchema);

